account_payment_return/models/payment_return.py

```diff
@@ -0,0 +1,193 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api, _
+from openerp.exceptions import except_orm
+
+
+class PaymentReturn(models.Model):
+    _name = "payment.return"
+    _inherit = ['mail.thread']
+    _description = 'Payment return'
+    _order = 'date DESC, id DESC'
+
+    company_id = fields.Many2one(
+        'res.company', 'Company', required=True,
+        help="Company",
+        states={'done': [('readonly', True)],
+                'cancelled': [('readonly', True)]
+                },
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account'))
+    date = fields.Date('Return date',
+                       help="This date will be used as the date for the "
+                       "account entry.",
+                       states={'done': [('readonly', True)],
+                               'cancelled': [('readonly', True)]},
+                       default=lambda *x: fields.Date.today())
+    name = fields.Char(
+        "Reference", size=64, required=True,
+        states={'done': [('readonly', True)],
+                'cancelled': [('readonly', True)]},
+        default=lambda self: self.env['ir.sequence'].next_by_code(
+            'payment.return'))
+    period_id = fields.Many2one('account.period', 'Forced period',
+                                states={'done': [('readonly', True)],
+                                        'cancelled': [('readonly', True)]
+                                        })
+    line_ids = fields.One2many('payment.return.line', 'return_id',
+                               states={'done': [('readonly', True)],
+                                       'cancelled': [('readonly', True)]
+                                       })
+    journal_id = fields.Many2one('account.journal', 'Bank journal',
+                                 required=True,
+                                 states={'done': [('readonly', True)],
+                                         'cancelled': [('readonly', True)]
+                                         })
+    move_id = fields.Many2one('account.move',
+                              'Reference to the created journal entry',
+                              states={'done': [('readonly', True)],
+                                      'cancelled': [('readonly', True)]
+                                      })
+    notes = fields.Text('Notes')
+    state = fields.Selection([('draft', 'Draft'),
+                              ('imported', 'Imported'),
+                              ('done', 'Done'),
+                              ('cancelled', 'Cancelled')],
+                             'State', readonly=True, default='draft',
+                             track_visibility='onchange')
+
+    @api.multi
+    def unlink(self):
+        attachment_obj = self.env['ir.attachment']
```

Comment:
I think in v8 this is done by default, so there's no need of this, but please check.